PlutoHelperAgent/AppDelegate.m

```diff
@@ -125,92 +125,150 @@ -    (void)getUrl:(NSAppleEventDescriptor *)event
             [alert runModal];
         };
         
-        [ProjectLockerAndKeychainFunctions get_data_from_server:@"/api/project/"
-                                                               :@"/files"
-                                                               :projectid
-                                              completionHandler:^void (NSURLResponse *response,NSDictionary *filesResult){
-            [ProjectLockerAndKeychainFunctions get_data_from_server:@"/api/storage/"
-                                                                   :NULL
-                                                                   :filesResult[@"files"][0][@"storage"]
-                                                  completionHandler:^(NSURLResponse *response, NSDictionary *storageResult) {
-                if (storageResult[@"result"][@"clientpath"] == NULL) {
-                    NSAlert *alert = [[NSAlert alloc] init];
-                    
-                    [alert addButtonWithTitle:@"Okay"];
-                    
-                    NSString *message = [NSString stringWithFormat: @"No client path on storage ID %@", filesResult[@"files"][0][@"storage"]];
-                    
-                    [alert setMessageText:message];
-                    
-                    [alert setInformativeText:@"Can't open project, because it's on a storage which has no client path set.\n\nPlease contact multimediatech@theguardian.com."];
-                    
-                    [alert setAlertStyle:NSWarningAlertStyle];
-                    
-                    if ([alert runModal] == NSAlertFirstButtonReturn) {
-                        
-                    }
-                    
-                } else {
-                    NSAlert *alert = [[NSAlert alloc] init];
-                    NSString *helperScript = [[NSUserDefaults standardUserDefaults] stringForKey:@"local_shell_script"];
-                    
-                    if([helperScript compare:@""]==0){
-                        dispatch_async(dispatch_get_global_queue(0, 0), ^{
-                            [alert setMessageText:@"Setup problem"];
-                            [alert setInformativeText:@"Your mac does not appear to be set up correctly, no helper script is configured. Please contact multimediatech@theguardian.com."];
+        [ProjectLockerAndKeychainFunctions check_logged_in:^(enum ReturnValues connectionStatus) {
+            
+            if (connectionStatus == ALLOK) {
```

Comment:
in Cocoa and other C type languages this set of blocks would be better off as a switch() rather than a set of ifs

Reply:
it's just that Python is a bit rubbish in that it does not have a switch() equivalent statement